src/databases/CGNS/avtCGNSFileReader.C

```diff
@@ -1625,6 +1629,151 @@ avtCGNSFileReader::GetCoords(int timestate, int base, int zone, const cgsize_t *
     return !err;
 }
 
+// ****************************************************************************
+// Method: avtCGNSFileReader::GetQuadGhostZones
+//
+// Purpose:
+//   Add the ghost zone information to the specified data set.
+//
+// Arguments:
+//   base       : The base to use
+//   zone       : The zone (mesh) that whose rind data we want.
+//   zsize      : The zone size information.
+//   cell_dim   : The cell dimension.
+//   ds         : The data set to add the ghost data to.
+//
+// Programmer: Eric Brugger
+// Creation:   Tue Jul  6 10:27:03 PDT 2021
+//
+// Modifications:
+//
+// ****************************************************************************
+
+void
+avtCGNSFileReader::GetQuadGhostZones(int base, int zone,
+    const cgsize_t *zsize, int cell_dim, vtkDataSet *ds)
+{
+    const char *mName = "avtCGNSFileReader::GetQuadGhostZones: ";
+
+    // Read the rind data.
+    debug4 << "Reading rind node." << endl;
+    int rind[6];
+    if(cg_goto(GetFileHandle(), base, "Zone_t", zone, "FlowSolution_t", 1, "end") == CG_OK)
+    {
+        if (cg_rind_read(rind) == CG_OK)
+        {
+            debug4 << "rind=" << rind[0] << "," << rind[1] << ","
+                              << rind[2] << "," << rind[3] << ","
+                              << rind[4] << "," << rind[5] << endl;
+        }
+        else
+        {
+            debug4 << "No rind data." << endl;
+            return;
+        }
+    }
+    else
+    {
+        debug4 << "Error going to FlowSolution." << endl;
+        return;
+    }
+
+    // Determine the size of the mesh.
+    unsigned int ncells = 0;
+    cgsize_t cdims[3] = {1,1,1};
+    if(cell_dim == 1)
+    {
+        cdims[0] = zsize[0]-1;
+    }
+    else if(cell_dim == 2)
+    {
+        cdims[0] = zsize[0]-1;
+        cdims[1] = zsize[1]-1;
+    }
+    else
+    {
+        cdims[0] = zsize[0]-1;
+        cdims[1] = zsize[1]-1;
+        cdims[2] = zsize[2]-1;
+    }
+    ncells = cdims[0] * cdims[1] * cdims[2];
+
+    // Determine if the rind information is valid.
+    cgsize_t first[3];
+    cgsize_t last[3];
+    bool ghostPresent = false;
+    bool badIndex = false;
+    for (int i = 0; i < cell_dim; i++)
+    {
+        first[i] = rind[i*2];
+        last[i]  = cdims[i] - rind[i*2+1] - 1;
+
+        if (first[i] < 0 || first[i] >= cdims[i])
+        {
+            debug1 << "bad Index on first[" << i << "] dims is: "
```

Comment:
Should the word `rind` appear in any of these error messages?

Reply:
Improved the error message.

Reply:
Improved the error message.